.github/workflows/cloudformation.yml

```diff
@@ -0,0 +1,153 @@
+        while [ "$STATUS" == "CREATE_IN_PROGRESS" ]
+        do
+          echo "Getting change set..."
+          CHANGE_SET=$(aws cloudformation describe-change-set \
+          --stack-name ${{ inputs.stack_name }} \
+          --change-set-name ${{ inputs.change_set_name }})
+          STATUS=$(jq -r '.["Status"]' <<< "$CHANGE_SET")
+        done
+        echo $CHANGE_SET
+        STACK_NAME=$(jq -r '.["StackName"]' <<< "$CHANGE_SET")
+        STACK_ID=$(jq -r '.["StackId"]' <<< "$CHANGE_SET")
+        CHANGE_SET_NAME=$(jq -r '.["ChangeSetName"]' <<< "$CHANGE_SET")
+        CHANGE_SET_ID=$(jq -r '.["ChangeSetId"]' <<< "$CHANGE_SET")
+        STATUS_REASON=$(jq -r '.["StatusReason"]' <<< "$CHANGE_SET")
+        PARAMETERS=$(jq '.["Parameters"]' <<< "$CHANGE_SET")
+        CAPABILITIES=$(jq '.["Capabilities"]' <<< "$CHANGE_SET")
+        CHANGES=$(jq '.["Changes"]' <<< "$CHANGE_SET")
+        echo "## [${{inputs.target_environment}}] Change Set for ${{ inputs.stack_name }}" >> $GITHUB_STEP_SUMMARY
+        echo "- **Stack Name:** $STACK_NAME" >> $GITHUB_STEP_SUMMARY
+        echo "- **Change Set Name:** $CHANGE_SET_NAME" >> $GITHUB_STEP_SUMMARY
+        echo "- **Stack ID:** $STACK_ID" >> $GITHUB_STEP_SUMMARY
+        echo "- **Change Set ID:** $CHANGE_SET_ID" >> $GITHUB_STEP_SUMMARY
+        echo "" >> $GITHUB_STEP_SUMMARY
+        echo "#### Parameters:" >> $GITHUB_STEP_SUMMARY
+        echo "~~~ json"  >> $GITHUB_STEP_SUMMARY
+        echo "$PARAMETERS" >> $GITHUB_STEP_SUMMARY
+        echo "~~~"  >> $GITHUB_STEP_SUMMARY
+        echo "#### Capabilities:" >> $GITHUB_STEP_SUMMARY
+        echo "~~~ json"  >> $GITHUB_STEP_SUMMARY
+        echo "$CAPABILITIES" >> $GITHUB_STEP_SUMMARY
+        echo "~~~"  >> $GITHUB_STEP_SUMMARY
+        echo "#### Changes:" >> $GITHUB_STEP_SUMMARY
+        echo "~~~ json"  >> $GITHUB_STEP_SUMMARY
+        if [ "$STATUS" == "FAILED" ]; then
+          echo "$STATUS_REASON" >> $GITHUB_STEP_SUMMARY
+        else
+          echo "$CHANGES" >> $GITHUB_STEP_SUMMARY
+        fi
+        echo "~~~"  >> $GITHUB_STEP_SUMMARY
+  
+  execute_change_set:
+    name: Execute Change Set
+    if: ${{ inputs.execute_change_set == true }}
+    needs: create_change_set
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.target_environment }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: eu-west-2
+          role-to-assume: ${{ secrets.cf_deploy_role }}
+          role-session-name: github-actions
+
+      - name: Execute Change Set
+        shell: bash
+        run: |
+          aws cloudformation execute-change-set \
+          --stack-name ${{ inputs.stack_name }} \
+          --change-set-name ${{ inputs.change_set_name }}
+
+  delete_change_set:
+    name: Delete Change Set
+    if: ${{ inputs.execute_change_set == false || cancelled() || failure() }}
```

Comment:
do we want the change set deleted if there is a failure? might be better if its kept in case of problems making it easier to diagnose